src/VisualStudio/Core/Def/Implementation/Watson/WatsonTraceListener.cs

```diff
@@ -4,13 +4,10 @@
 
 using System;
 using System.Diagnostics;
-using Microsoft.CodeAnalysis.ErrorReporting;
 
-namespace Microsoft.VisualStudio.IntegrationTest.Setup
+namespace Microsoft.CodeAnalysis.ErrorReporting
 {
-    using Debugger = System.Diagnostics.Debugger;
-
-    internal class IntegrationTestTraceListener : TraceListener
+    internal class WatsonTraceListener : TraceListener
     {
         public override void Fail(string message, string detailMessage)
```

Comment:
>Fail [](start = 29, length = 4)

How does this behave when the debugger is attached? When debugging we shouldn't just fail fast on failed assert.

Reply:
This code just delegates to `FatalError.Report`. Handling for debugger attachment, if any, would be there.

Also note that attaching the debugger to the ServiceHub process is not a standard scenario.

Reply:
> Also note that attaching the debugger to the ServiceHub process is not a standard scenario.

I do it all the time. How else would you debug our remote services?